labs-ios-starter/Surveys/Model Controllers/TopicController.swift

```diff
@@ -0,0 +1,179 @@
+    //
+//  TopicController.swift
+//  labs-ios-starter
+//
+//  Created by Kenny on 9/12/20.
+//  Copyright © 2020 Lambda, Inc. All rights reserved.
+//
+
+import Foundation
+
+/// Controller for Topic, ContextObject, and Question model
+class TopicController {
+    let networkService = NetworkService.shared
+    let profileController = ProfileController.shared
+    lazy var baseURL = profileController.baseURL
+
+    ///public getter for CONTEXTS
+    var contexts: [ContextObject]  {
+        CONTEXTS
+    }
+
+    ///private setter for contexts
+    private var CONTEXTS: [ContextObject] = []
+
+    var questions: [Question] {
+        QUESTIONS
+    }
+
+    private var QUESTIONS: [Question] = []
+
+    typealias CompleteWithNeworkError = (Result<Void, ErrorHandler.NetworkError>) -> Void
+
+    private func createRequest(auth: Bool = true,
+                               pathFromBaseURL: String,
+                               method: NetworkService.HttpMethod = .get) -> URLRequest? {
+
+        let targetURL = baseURL.appendingPathComponent(pathFromBaseURL)
+
+        guard var request = networkService.createRequest(url: targetURL, method: method, headerType: .contentType, headerValue: .json) else {
+            print("unable to create request for \(targetURL)")
+            return nil
+        }
+
+        if auth {
+            //add bearer to request
+            request.addAuthIfAvailable()
+        }
+
+        return request
+    }
+
+    typealias CompleteWithTopics = (Result<[Topic], ErrorHandler.NetworkError>) -> Void
+
+    ///fetch all topics, or topics for the logged in user
+    func fetchTopic(all: Bool = true, completion: @escaping CompleteWithTopics) {
+        var appendToURL = "topic"
+        if !all {
+            // Doesn't look like this will work - topic by id is topic id
+            // TODO: coordinate with BE team to see if we can get a topic by userId endpoint
+            appendToURL = "\(ProfileController.shared.authenticatedUserProfile!.id!)"
+        }
```

Comment:
Is this going to cause any bugs?
```suggestion
//        if !all {
            // Doesn't look like this will work - topic by id is topic id
            // TODO: coordinate with BE team to see if we can get a topic by userId endpoint
//            appendToURL = "\(ProfileController.shared.authenticatedUserProfile!.id!)"
        }
```

Reply:
It'll be fine once BE implements that endpoint. That said, the force unwrap will go away once its implemented